src/main/java/bot/assembly/function/BotCommandResponderUnit.java

```diff
@@ -0,0 +1,303 @@
+package bot.assembly.function;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import bot.assembly.error.InvalidCommandException;
+import bot.assembly.error.InvalidCommandFormatException;
+import bot.assembly.error.InvalidTaskIndexException;
+import bot.assembly.error.TaskOutOfRangeException;
+import bot.assembly.memory.BotDynamicMemoryUnit;
+import bot.assembly.memory.BotStaticMemoryUnit;
+import bot.assembly.memory.CommandInput;
+import bot.assembly.task.Deadline;
+import bot.assembly.task.Event;
+import bot.assembly.task.Task;
+import bot.assembly.task.ToDo;
+
+/**
+ * A class that handles the command input of the user
+ */
+public class BotCommandResponderUnit {
```

Comment:
This is a huge class that does many things, is it possible to let some of the components handle their own responsibilities rather than having them all be done by this class?